src/test/ui/lint/use-redundant.rs

```diff
@@ -0,0 +1,17 @@
+// compile-pass
+
+use crate::foo::Bar; //~ WARNING first import
+
+mod foo {
+    pub type Bar = i32;
+}
+
+fn baz() -> Bar {
+    3
+}
+
+fn main() {
+    use crate::foo::Bar; //~ WARNING redundant import
+    let _a: Bar = 3;
+    baz();
+}
```

Comment:
Could you add this test case:
```rust
mod m1 { struct S {} }
mod m2 { struct S {} }

use m1::*;
use m2::*;

fn main() {
    use m1::S;
    let s = S;
}
```

Reply:
This import is not really redundant, but the lint as it's implemented now will report a warning.
We'll need to find a way to ignore this case.